support/cas-server-support-cassandra-ticket-registry/src/main/java/org/apereo/cas/ticket/registry/CassandraTicketRegistry.java

```diff
@@ -0,0 +1,235 @@
+        final long currentTime = currentTimeBucket();
+        if (lastRun == 0 || lastRun > currentTime) {
+            updateLastRunTimestamp(currentTime);
+        }
+    }
+
+    @Override
+    public void addTicket(final Ticket ticket) {
+        final String ticketId = ticket.getId();
+        LOGGER.debug("Inserting ticket {}", ticketId);
+        final TicketDefinition ticketDefinition = ticketCatalog.find(ticketId);
+        final String storageName = ticketDefinition.getProperties().getStorageName();
+
+        if (tgtTable.equals(storageName)) {
+            final String parentTgtId = ticket.getGrantingTicket() == null ? null : ticket.getGrantingTicket().getId();
+            session.execute(this.insertTgtStmt.bind(ticket.getId(), serializer.serialize(ticket), parentTgtId, calculateExpirationDate(ticket) / TEN));
+        } else if (stTable.equals(storageName)) {
+            session.execute(this.insertStStmt.bind(ticket.getId(), serializer.serialize(ticket)));
+        } else {
+            LOGGER.error("Failed to insert ticket type {}", ticket.getClass().getName());
+        }
+    }
+
+    @Override
+    public long deleteAll() {
+        return TICKETS_IN_TESTS;
+    }
+
+    @Override
+    public boolean deleteSingleTicket(final String ticketId) {
+        LOGGER.debug("Deleting ticket {}", ticketId);
+        if (ticketId == null) {
+            return false;
+        }
+        final TicketDefinition ticketDefinition = ticketCatalog.find(ticketId);
+        final String storageName = ticketDefinition.getProperties().getStorageName();
+        if (tgtTable.equals(storageName)) {
+            return session.execute(this.deleteTgtStmt.bind(ticketId)).wasApplied();
+        } else if (stTable.equals(storageName)) {
+            session.executeAsync(this.deleteStStmt.bind(ticketId));
+            return true;
+        } else {
+            LOGGER.error("Failed to delete ticket type {}", ticketId);
+            return false;
+        }
+    }
+
+    @Override
+    public Ticket getTicket(final String ticketId) {
+        LOGGER.debug("Querying ticket {}", ticketId);
+        if (ticketId == null) {
+            return null;
+        }
+        final TicketDefinition ticketDefinition = ticketCatalog.find(ticketId);
+        if (ticketDefinition == null) {
+            return null;
+        }
+        final PreparedStatement statement = getTicketQueryForStorageName(ticketDefinition);
+        final Row row = session.execute(statement.bind(ticketId)).one();
+        if (row == null) {
+            LOGGER.info("ticket {} not found", ticketId);
+            return null;
+        }
+        return serializer.deserialize(row.getString(FIRST_COLUMN_INDEX), ticketDefinition.getImplementationClass());
+    }
+
+    private PreparedStatement getTicketQueryForStorageName(final TicketDefinition ticketDefinition) {
+        final String storageName = ticketDefinition.getProperties().getStorageName();
```

Comment:
This sort of logic here needs to be avoided. It defeats the polymorphic functionality of the catalog. Any time you find yourself checking for a ticket type to do something, that's a bad sign and needs to be reviewed.